packages/core/src/use-router-query.ts

```diff
@@ -1,8 +1,18 @@
+import {useRef} from "react"
+import {isEqual} from "lodash"
 import {useRouter} from "next/router"
 import {parse} from "url"
 
 export function useRouterQuery() {
   const router = useRouter()
+
   const {query} = parse(router.asPath, true)
-  return query
+
+  const lastQueryRef = useRef(query)
+
+  if (!isEqual(query, lastQueryRef.current)) {
+    lastQueryRef.current = query
+  }
+
+  return lastQueryRef.current
```

Comment:
I'm not confident that this is concurrent mode safe. I've asked for confirmation [on twitter](https://twitter.com/flybayer/status/1310599536964968454).

Reply:
No this isn't CM safe. https://twitter.com/adamrackis/status/1309665030967889920?s=21

Can only write to the ref inside useEffect